src/battle_ai_switch_items.c

```diff
@@ -1785,14 +1785,65 @@ static bool32 CanAbilityTrapOpponent(u16 ability, u32 opponent)
         return FALSE;
 }
 
+static bool32 IsFreeSwitch(bool32 isSwitchAfterKO, u32 battlerSwitchingOut, u32 opposingBattler)
+{
+    u32 i;
+    bool32 movedSecond = GetBattlerTurnOrderNum(battlerSwitchingOut) > GetBattlerTurnOrderNum(opposingBattler) ? TRUE : FALSE;
+    
+    // Switch out effects
+    if (!IsDoubleBattle()) // Not handling doubles' additional complexity
+    {
+        if (IsSwitchOutEffect(gMovesInfo[gLastUsedMove].effect) && movedSecond)
+            return TRUE;
+        if (gSpecialStatuses[battlerSwitchingOut].ejectButtonSwitch)
+            return TRUE;
+        if (gSpecialStatuses[battlerSwitchingOut].ejectPackSwitch)
+        {
+            // If faster and lowered own stat, not a free switch
+            if (!movedSecond)
+            {
+                for (i = 0; i < gMovesInfo[gLastUsedMove].numAdditionalEffects; i++)
+                {
+                    if (IsSelfStatLoweringEffect(gMovesInfo[gLastUsedMove].additionalEffects[i].moveEffect) && gMovesInfo[gLastUsedMove].additionalEffects[i].self)
+                        return FALSE;
+                }
+            }
+            // Otherwise, free switch
+            return TRUE;
+        }
```

Comment:
Oh good catch I missed a `return TRUE` after that loop / extra check.

The idea here is that if the AI moved second, it's always a free switch.

If the AI is moving first and they lowered their own stats, it's not a free switch.

If the AI is moving first and they did *not* lower their own stats, then something like Intimidate has procc'd and it's still a free switch in spite of their moving first. This is missing that `return TRUE`.

Does that logic seem sensical to you before I add that, or would you scrap the extra check?

Reply:
Also what d'you mean by taking into account status moves? :)

Reply:
Ah, I see. Though I still don't understand why we can't just use the turn order? It feels a little hacky to check a move that might have lowered stats.

Reply:
Can confirm this works the way I want it to with the much cleaner

```
        if (gSpecialStatuses[battlerSwitchingOut].ejectPackSwitch)
        {
            // If faster, not a free switch; likely lowered own stats
            if (!movedSecond)
                return FALSE;
            // Otherwise, free switch
            return TRUE;
        }
```

Will include the new test along with the rest of the comments!